Sources/ProjectSpec/TargetSource.swift

```diff
@@ -100,12 +142,42 @@ extension TargetSource: JSONObjectConvertible {
         excludes = jsonDictionary.json(atKeyPath: "excludes") ?? []
         type = jsonDictionary.json(atKeyPath: "type")
         optional = jsonDictionary.json(atKeyPath: "optional") ?? false
+        
         if let string: String = jsonDictionary.json(atKeyPath: "buildPhase") {
-            if let buildPhase = BuildPhase(rawValue: string) {
-                self.buildPhase = buildPhase
-            } else {
-                throw SpecParsingError.unknownSourceBuildPhase(string)
-            }
+            buildPhase = try BuildPhase(string: string)
+        } else if let dict: JSONDictionary = jsonDictionary.json(atKeyPath: "buildPhase") {
+            buildPhase = try BuildPhase(jsonDictionary: dict)
+        }
+    }
+}
+
+extension TargetSource.BuildPhase {
+    
+    public init(string: String) throws {
+        switch string {
+        case "sources": self = .sources
+        case "headers": self = .headers
+        case "resources": self = .resources
+        case "copyFiles":
+            throw SpecParsingError.invalidSourceCopyFilesPhase
+        case "none": self = .none
+        default:
+            throw SpecParsingError.unknownSourceBuildPhase(string)
         }
     }
 }
+
+extension TargetSource.BuildPhase: JSONObjectConvertible {
+    
+    public init(jsonDictionary: JSONDictionary) throws {
+        self = .copyFiles(try jsonDictionary.json(atKeyPath: "copyFiles"))
+    }
+}
+
+extension TargetSource.BuildPhase.CopyFilesSettings: JSONObjectConvertible {
+    
+    public init(jsonDictionary: JSONDictionary) throws {
+        destination = try jsonDictionary.json(atKeyPath: "destination")
+        subpath = jsonDictionary.json(atKeyPath: "subpath") ?? ""
```

Comment:
Would the default be nil?

Reply:
I'll adjust the docs to say it's optional. We should still default it to `""` though to match Xcode behavior (there isn't a way to make it not `""` by using just Xcode).